wagtail/documents/views/bulk_actions/add_to_collection.py

```diff
@@ -0,0 +1,55 @@
+from django import forms
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext
+
+from wagtail.core import hooks
+from wagtail.core.models.collections import Collection
+from wagtail.documents.permissions import permission_policy
+from wagtail.documents.views.bulk_actions.document_bulk_action import DocumentBulkAction
+
+
+class CollectionForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['collection'] = forms.ModelChoiceField(
+            queryset=Collection.objects.all()
+        )
+
+
+class AddToCollectionBulkAction(DocumentBulkAction):
+    display_name = _("Add to collection")
+    action_type = "add_to_collection"
+    aria_label = _("Add documents to collection")
+    template_name = "wagtaildocs/bulk_actions/confirm_bulk_add_to_collection.html"
+    action_priority = 30
+
+    def check_perm(self, document):
+        return permission_policy.user_has_permission_for_instance(self.request.user, 'change', document)
+
+    def execute_action(cls, documents):
+        collection_id = int(cls.request.POST.get('collection'))
```

Comment:
I feel like we should start working more with Django's form system, using `is_valid()` and the like and then retrieving data rather than getting it directly from `request.POST` - it lets us handle errors a bit better and return an invalid form to the user.

Reply:
For example:
```
form = CollectionForm(request.POST) # This gets you a form with all of the data in
if form.is_valid(): # Check if the form is valid
    collection_id = form.cleaned_data['collection'] # retrieve data from form.cleaned_data instead
    # now do other stuff
else:
    # return a response with the form in context, render its errors in the template
```

To avoid boilerplate, you may find `FormView` or similar handy, see https://docs.djangoproject.com/en/3.2/topics/class-based-views/generic-editing/

In your particular case, you may need to use `user` in the form to construct the set of collections. You can make the constructor accept `user` as well to do this.

Reply:
> To avoid boilerplate, you may find FormView or similar handy, see https://docs.djangoproject.com/en/3.2/topics/class-based-views/generic-editing/

Should this be changed at the base bulk action class, where we can add the `FormView` as another base class in addition to the `TemplateView`? Because more or less all actions have at least one input element

Reply:
If you end up finding `FormView` useful then yes, I'd suggest it adding it at the base class level.